src/theory/bags/solver_state.h

```diff
@@ -31,24 +31,43 @@ class SolverState : public TheoryState
  public:
   SolverState(context::Context* c, context::UserContext* u, Valuation val);
 
-  void registerClass(TNode n);
+  void registerBag(TNode n);
 
-  Node registerBagElement(TNode n);
-
-  std::set<Node>& getBags();
-
-  std::set<Node>& getElements(TypeNode t);
-
-  std::map<Node, Node>& getBagElements(Node B);
+  /**
+   * @param n has the form (bag.count e A)
+   * @pre bag A needs is already registered using registerBag(A)
+   * @return a unique skolem for (bag.count e A)
+   */
+  void registerCountTerm(TNode n);
+  /** get all bag terms */
```

Comment:
Can you be a bit more precise?  Is this the set of representatives? Or the set of all bag terms?  

Also, is this call only valid after the state has been initialized on a FULL effort check?

Reply:
Done. `SolverState::getBags()` returns only representatives in the equality engine. The function is only valid after initialization during `postCheck`.